### PR TITLE
Implement debugloop

### PR DESCRIPTION
**Pull request by matthewturk**
This is a port of https://github.com/caiiiycuk/js-dos/pull/158 to the new repository.

It fixes the address of registers and segments, and it also adds a dosbox-level debug loop.  This allows us to continue to call dosbox-internal functions without stopping the emscripten main loop.

### CODE REVIEW

File: native/dosbox-jsdos/jsdos/include/jsdos-timer.h [hunk 1]
```diff
@@ -9,4 +9,6 @@ namespace jsdos {
   void initTimer();
 }
 
+void DelayWithYield(int);
```
Comment (caiiiycuk) [inline native/dosbox-jsdos/jsdos/include/jsdos-timer.h:12]: Do you need it? I don't see where it's called...

File: src/dos/dosbox/cpp/direct-debugger.cpp [hunk 2]
```diff
@@ -67,4 +157,23 @@ extern "C" void EMSCRIPTEN_KEEPALIVE rescanFilesystem() {
   return;
 }
 
+Bitu DosBox_Pause(void) {
+  asyncify_sleep(1);
+  return 0;
+}
+
+extern "C" bool EMSCRIPTEN_KEEPALIVE pauseExecution(bool shouldPause) {
```
Comment (caiiiycuk) [inline src/dos/dosbox/cpp/direct-debugger.cpp:165]: please add information about this feature to wiki
https://github.com/caiiiycuk/js-dos/blob/gh-pages/v7/docs/dosbox-direct.md

**Review by caiiiycuk — CHANGES_REQUESTED** [review]

File: native/dosbox-jsdos/jsdos/include/jsdos-timer.h (hunk 1 above)
Comment (matthewturk) [reply native/dosbox-jsdos/jsdos/include/jsdos-timer.h:12]: You're right -- thank you.

File: src/dos/dosbox/cpp/direct-debugger.cpp [hunk 3]
```diff
@@ -67,4 +157,23 @@ extern "C" void EMSCRIPTEN_KEEPALIVE rescanFilesystem() {
   return;
 }
 
+Bitu DosBox_Pause(void) {
```
[the remark this answers is not in the document]
Comment (matthewturk) [reply src/dos/dosbox/cpp/direct-debugger.cpp:160]: Done!  I've put it in alpha order.  I think I assumed this was auto-generated.

File: src/dos/dosbox/cpp/direct-debugger.cpp (hunk 2 above)
Comment (matthewturk) [reply src/dos/dosbox/cpp/direct-debugger.cpp:165]: Will do.

**Pull request merged**